lib/util/http/index.ts

```diff
@@ -198,11 +196,25 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
           )
         : null;
 
+    if (memCacheKey) {
+      cacheProvider = undefined;
+    }
+
+    const cachedResponse = await cacheProvider?.bypassServer<T>(url);
+    if (cachedResponse) {
+      return cachedResponse;
+    }
+
     let resPromise: Promise<HttpResponse<T>> | null = null;
 
     // Cache GET requests unless memCache=false
     if (memCacheKey) {
       resPromise = memCache.get(memCacheKey);
+
+      // istanbul ignore next
+      if (resPromise && options.cacheProvider !== memCacheProvider) {
+        logger.debug({ url }, 'Cache hit on the obsolete memCache option');
```

Comment:
I switched this to logger.error and ran on containerbase/base with dryRun=lookup and got many errors. They're all either `/v2/` queries or `/token?` ones from docker datasource.

Maybe this datasource should do its _own_ caching of these, with its own TTLs?

Reply:
I think I'd prefer to accumulate these URLs into a Set and then at the end print a debug message about number of URLs using legacy caching. Then we can capture and graph that in our hosted app and know when we reach 0